From a2154c08893ddcadc25e6987595d2ccede182341 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Mon, 4 Dec 2017 16:45:13 +0800
Subject: [PATCH 2113/2241] doc: ata: add ahci-mvebu dt-bindings documentation

Marvell EBU AHCI SATA Controller is a kind of platform AHCI SATA
Controller, but it supports software activity LED blinking through
a GPIO when the disk is connected to the controller directly.

Change-Id: I3d3115941ebae75a914092a5146b8acddb5e89ba
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47188
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 .../devicetree/bindings/ata/ahci-mvebu.txt         | 63 ++++++++++++++++++++++
 1 file changed, 63 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/ata/ahci-mvebu.txt

diff --git a/Documentation/devicetree/bindings/ata/ahci-mvebu.txt b/Documentation/devicetree/bindings/ata/ahci-mvebu.txt
new file mode 100644
index 0000000..8b7d2db
--- /dev/null
+++ b/Documentation/devicetree/bindings/ata/ahci-mvebu.txt
@@ -0,0 +1,63 @@
+* Marvell EBU AHCI SATA Controller
+
+Marvell EBU AHCI SATA Controller belongs to platform AHCI SATA
+Controller; for more detailed about platform AHCI, please refer
+to "ahci-platform.txt".
+
+Required properties:
+- compatible        : compatible string, one of:
+  - "marvell,armada-380-ahci"
+  - "marvell,armada-3700-ahci"
+  - "marvell,armada-cp110-ahci"
+- interrupts        : <interrupt mapping for SATA IRQ>
+- reg               : <registers mapping>
+
+Optional properties in addition to platform AHCI:
+- "marvell,led-gpio"          : the GPIO which controlls LED state
+- "marvell,blink-policy"      : LED blinking policy - 0 the LED is not activated on activity
+                                                      1 the LED blinks on every 10ms when activity is detected.
+                                                      2 the LED is on when idle, and blinks off every 10ms when activity is detected.
+LED Blinking:
+Marvell EBU AHCI SATA Controller supports software activity LED
+blinking through a GPIO with the optional properties of "marvell,led-gpio"
+and "marvell,blink-policy". The software activity LED blinking only
+works when the SATA disk is connected to the controller directly,
+so the 2 optional properties should not be set when a PMP card is
+attached to the controller and disks are connected to the PMP card.
+
+Example for armada 3720 DB:
+        sata@e0000 {
+		compatible = "marvell,armada-3700-ahci";
+		reg = <0xe0000 0x178>;
+		interrupts = <GIC_SPI 27 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&nb_periph_clk 1>;
+		target-supply = <&reg_5v_sata>;
+		phys = <&a3700_comphy 2 COMPHY_SATA0>;
+		marvell,led-gpio = <&gpionb 11 GPIO_ACTIVE_LOW>;
+		marvell,blink-policy = <BLINK_OFF>;
+		status = "okay";
+	};
+
+Example for armada 8040 DB:
+	cp0_sata: sata@540000 {
+		compatible = "marvell,armada-cp110-ahci";
+		reg = <0x540000 0x30000>;
+		dma-coherent;
+		interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cp0_syscon0 1 15>;
+		comreset_u = <0x17>;
+		comwake = <0x6>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sata-port@0 {
+			reg = <0>;
+			phys = <&cp0_comphy 1 COMPHY_SATA0>;
+			phy-names = "comphy";
+		};
+		sata-port@1 {
+			reg = <1>;
+			phy-names = "comphy";
+			status = "okay";
+		};
+	};
\ No newline at end of file
-- 
2.7.4

